src/sage/modules/free_module_integer.py

```diff
@@ -867,12 +877,39 @@ def approximate_closest_vector(self, t, delta=None, *args, **kwargs):
             self.LLL(*args, delta=delta, **kwargs)
 
         B = self._reduced_basis
-        G = B.gram_schmidt()[0]
         t = vector(t)
 
-        b = t
-        for i in reversed(range(G.nrows())):
-            b -= B[i] * ((b * G[i]) / (G[i] * G[i])).round("even")
-        return (t - b).change_ring(ZZ)
+        if algorithm == 'embedding':
+            L = matrix(QQ, B.nrows()+1, B.ncols()+1)
+            L.set_block(0, 0, B)
+            L.set_block(B.nrows(), 0, matrix(t))
+            weight = (B[-1]*B[-1]).isqrt()+1 # norm of the largest vector
+            L[-1, -1] = weight
+
+            # the vector should be the last row but we iterate
+            # just in case
+            for v in reversed(L.LLL(delta=delta, *args, **kwargs).rows()):
+                if abs(v[-1]) == weight:
+                    return t - v[:-1]*v[-1].sign()
+            raise ValueError('No suitable vector found in basis.'
+                             'This is a bug, please report it.')
+
+        elif algorithm == 'nearest_plane':
+            G = B.gram_schmidt()[0]
+
+            b = t
+            for i in reversed(range(G.nrows())):
+                b -= B[i] * ((b * G[i]) / (G[i] * G[i])).round("even")
+            return (t - b).change_ring(ZZ)
+
+        elif algorithm == 'rounding_off':
+            # t = x*B might not have a solution over QQ so we instead solve
+            # the system x*B*B^T = t*B^T which will be the "closest" solution
+            # if it does not exist, same effect as using the psuedo-inverse
+            return vector(ZZ,
+                [QQ(x).round('even') for x in (B*B.T).solve_left(t*B.T)])*B
+
+        else:
+            raise ValueError("algorithm must be one of 'embedding', 'nearest_plane' or 'rounding_off'")
 
-    babai = approximate_closest_vector
\ No newline at end of file
+    cvp = approximate_closest_vector
```

Comment:
This will need deprecation

Reply:
Is this because there has been a release candidate since this functions introduction or do we generally require deprecations for changes within the beta stage?

Reply:
Yes, it's unlikely that this PR will make it in the 10.4 release. The release manager typically only merges "blocker" PRs in this stage.

Reply:
Gotcha, how does one deprecate an alias like this one? I also see now that the alias gets its own entry in the reference which feels a little superfluous.

Reply:
Okay I opted for not changing the alias since it was a minor nitpick anyway, I think "babai" is reasonable (all the algorithms arguably in some way stem from him). I also added a small function body around the alias which redirects to the main one so as not to duplicate the entire docstring in the reference manual.